src/com/trilead/ssh2/util/TimeoutService.java

```diff
@@ -20,25 +21,33 @@
  * @version $Id: TimeoutService.java,v 1.1 2007/10/15 12:49:57 cplattne Exp $
  */
 public class TimeoutService {
+    
+    
+    private  ScheduledFuture<?> scheduledFuture;
+    private final String hostname;
+    private final ThreadFactory threadFactory = new ThreadFactory() {
 
-    private final static ThreadFactory threadFactory = new ThreadFactory() {
-
-        private AtomicInteger count = new AtomicInteger();
-
+        @Override
         public Thread newThread(Runnable r) {
-            int threadNumber = count.incrementAndGet();
-            String threadName = "TimeoutService-" + threadNumber;
+            
+            String threadName = hostname+"_Trilead_TimeoutService";
```

Comment:
"Trilead_TimeoutService"+hostname
to better align with other thread names like https://github.com/jenkinsci/trilead-ssh2/pull/135/files